static/screeningsForStartpage.js

```diff
@@ -0,0 +1,118 @@
+export async function renderUpcomingScreenings() {
+  const res = await fetch('http://localhost:5080/api/screenings');
+  const payload = await res.json();
+
+  // Create headline for "nästkommande visningar"
+  const upcomingMovies = document.querySelector('.upcomingMovies');
+  const screeningsDOMHeadline = document.createElement('h2');
+  screeningsDOMHeadline.innerText = 'Nästkommande visningar på Bio Regna';
+  upcomingMovies.append(screeningsDOMHeadline);
+  const screeningsDOMList = document.createElement('ul');
+  upcomingMovies.append(screeningsDOMList);
+
+  // Create array of months to use later for date categories
+  const months = [
+    'januari',
+    'februari',
+    'mars',
+    'april',
+    'maj',
+    'juni',
+    'juli',
+    'augusti',
+    'september',
+    'oktober',
+    'november',
+    'december',
+  ];
+
+  // Create date category array
+  const screeningDateCategories = [];
+
+  for (let i = 0; i < payload.length; i++) {
+    const screeningDate = new Date(payload[i].attributes.start_time);
+    const screeningDateArray = [
+      screeningDate.getDate(),
+      screeningDate.getMonth(),
+      screeningDate.getFullYear(),
+    ];
+    if (
+      JSON.stringify(
+        screeningDateCategories[screeningDateCategories.length - 1]
+      ) != JSON.stringify(screeningDateArray)
+    ) {
+      screeningDateCategories.push(screeningDateArray);
+    }
+  }
```

Comment:
The algorithm you have invented here (and in the rest of this module) is great. It basically uses a "dictionary" of known dates, which is fast to look up. But you've made some decisions that makes the code more difficult than it needs to be. For example, using a Set instead of an array would mean that you wouldn't have to worry about the if statement to keep things unique. And instead of using arrays of date, month and year here, the code would be much simpler if you just used a `YYYY-MM-DD` string, e.g:

```js
// A set is like an array, but every item must be unique
const screeningDateCategories = new Set();

for (let i = 0; i < payload.length; i++) {
  const startTimeString = payload[i].attributes.start_time;
  const dateString = startTimeString.slice(0, 10); // First 10 characters is YYYY-MM-DD
  screeningDateCategories.add(dateString); // Set keeps it unique
}
```

You would of course have to make some changes to the code below as well, but I believe that would also be simplified by this alternative data structure.